mkt/langpacks/models.py

```diff
@@ -81,21 +84,55 @@ def handle_file_operations(self, upload):
         """Handle file operations on an instance by using the FileUpload object
         passed to set filename, size, hash on the LangPack instance, and moving
         the temporary file to its final destination."""
-        # FIXME: store validation result ?
         upload.path = smart_path(nfd_str(upload.path))
         if not self.uuid:
             self.reset_uuid()
         self.filename = self.generate_filename()
```

Comment:
Ah, I didn't want to inline it to keep the API similar to `File` and keep the code readable. Now I actually use it in the tests to generate a real filename when creating fake data.